python/lsst/db/mysqlUtils.py

```diff
@@ -0,0 +1,387 @@
+            theKey = optionToConnectArgMap.get(o, o)
+            ret[theKey] = cnf.get(theSection, o)
+    logger.info("fetched %s from '%s' (password not shown)" % (
+            str(["%s:%s" % (x, ret[x]) for x in ret if not x == "passwd"]), fName))
+    return ret
+
+#### Database-related functions ####################################################
+def createDb(conn, dbName, mayExist=False):
+    """
+    Create database <dbName>.
+
+    @param dbName      Database name.
+    @param mayExist    Flag indicating what to do if the database exists.
+
+    Raises DbException if the database already exists and mayExist is False, or if
+    MySQL error occurs.
+
+    Note, it will not connect to that database and it will not make it default.
+    """
+    if dbName is None:
+        raise DbException(DbException.INVALID_DB_NAME, "<None>")
+    try:
+        conn.execute("CREATE DATABASE `%s`" % dbName)
+    except DBAPIError as e:
+        if e.orig[0] == MySqlErr.ER_DB_CREATE_EXISTS:
+            if not mayExist:
+                raise DbException(DbException.DB_EXISTS, dbName)
+        else:
+            raise DbException(DbException.MYSQL_ERROR, e.orig[0], e.orig[1])
+
+def useDb(conn, dbName):
+    """
+    Connect to database <dbName>.
+
+    @param dbName     Database name.
+
+    Raises DbException if MySQL error occurs.
+    """
+    if dbName is not None:
+        try:
+            conn.execute("USE `%s`" % dbName)
+        except DBAPIError as e:
+            if e.orig[0] == MySqlErr.ER_BAD_DB_ERROR:
+                raise DbException(DbException.DB_DOES_NOT_EXIST, dbName)
+            else:
+                raise DbException(DbException.MYSQL_ERROR, e.orig[0], e.orig[1])
+
+def dbExists(conn, dbName):
+    """
+    Return True if database <dbName> exists, False otherwise.
+
+    @param dbName     Database name.
+
+    Raises DbException if MySQL error occurs.
+    """
+    if dbName is None:
+        return False
+    cmd = "SELECT COUNT(*) FROM information_schema.schemata "
+    cmd += "WHERE schema_name = '%s'" % dbName
+    try:
+        return conn.execute(cmd).first()[0] == 1
+    except DBAPIError as e:
+        raise DbException(DbException.MYSQL_ERROR, e.orig[0], e.orig[1])
+
+def dropDb(conn, dbName, mustExist=True):
+    """
+    Drop database <dbName>.
+
+    @param dbName     Database name.
+    @param mustExist  Flag indicating what to do if the database does not exist.
+
+    Raises DbException if the database does not exist and the flag mustExist is
+    set to True, or if MySQL error occurs.
+
+    Disconnect from the database if it is the current database.
+    """
+    if not mustExist and not dbExists(conn, dbName):
+        return
+    try:
+        conn.execute("DROP DATABASE `%s`" % dbName)
+    except DBAPIError as e:
+        if e.orig[0] == MySqlErr.ER_DB_DROP_EXISTS:
+            raise DbException(DbException.DB_DOES_NOT_EXIST, dbName)
+        elif e.orig[0] == MySqlErr.ER_NO_DB_ERROR:
```

Comment:
Do you know when this error is generated? Looking at the codes the message for this error is "No database selected", I'm not sure it applies to DROP DATABASE.